### DISCUSSION TITLE
AnimationPlayer + Sprite should be its own node

### DISCUSSION THREAD

**Discussion by wyvernbw** [godotengine/godot-proposals#5662] [2D]
# Background info
I am working on a 2d game with complex animations and animation trees. I need things like collision detection to change within those animations and I currently achieve this with the AnimationPlayer node. For the sprite, I use an AnimatedSprite and call the `play` method within the animation player. But this makes timing the animation difficult. My other options are to use a normal Sprite and keyframe the `frame` property but this is a pain and when working with sprite sheets that are 60+ frames large this becomes unusable. 

# The problem
Not having a direct link between animation in sprites and AnimationPlayers. Getting the duration of each frame within the AnimationPlayer is quite difficult. 

# My solution
Basically just AnimatedSprite with a timeline that can do everything the normal AnimationPlayer can. Instead of moving on the timeline by 0.1s or whatever value you choose as your snap you can move frame by frame and you are able to select the duration of frames (think aseprite, maybe without the individual frame duration options).

# tl;dr
combination between AnimatedSprite and AnimationPlayer would be cool :)
I am curious how many people run into these issues, maybe it's just me

**Comment by dalexeev** [comment]
Related:

* godotengine/godot#65609

**Comment by wyvernbw** [comment]
OK so to better see what I meant look at this timeline from aseprite where each column is a frame in the animation. You can keyframe properties in these frames of animation. It should have the same the same sort of importing animations from spritesheets that AnimatedSprite has or something along those lines. So a way to have a direct and clear connection between frame x and what happens on frame x.

![image](https://user-images.githubusercontent.com/50150016/198398538-f4e4c6f9-f06c-45d9-a633-62928c496265.png)

Making this proposal I realize this would be a ton of work to implement. I know c++ quite well but I am not familiar with the engine's source code but I might try to familiarize myself with it and make a quick mockup
